### ISSUE TITLE
bundle pristine is overzealous with local overrides

### ISSUE THREAD

**Issue by jayjlawrence** [bundler/bundler#6836]
Your project includes a gem you're working on locally. So you make a local override for that dependent gem to allow you to work on the main project and use the local code for said gem.

You've been poking around in some other gem code to figure out what's going on. Put a bit of debug code here and there. You're done so you decide - bundle pristine - to bring all gems to, well pristine state.

WTF? Where'd my code I was working on go? Bundle pristine is so pristine that those local overrides need to go too. Ok, when I wanted pristine gems - I meant the ones that don't belong to me!

Steps to reproduce using Bundler 17.1 and Ruby 2.5.1 on Mac OS 10.14 are

```bash
# In a fresh directory:
bundle init
git clone git@github.com:codegram/cheezburger.git
echo 'gem "cheezburger", git: "git@github.com:codegram/cheezburger.git", branch: "master"' >>Gemfile
bundle config local.cheezburger
bundle install

# your own special spin on cheezburger specs now
# in your quest to really nail your plans you mucked with the sources for rspec etc ...
# what a mess!
# me hoeps u haz cheezburger now

# Had enough? Lets get back to reality ...
bundle pristine
```

Now I haz no cheezburger :( 

In other words - if the gem has been config'd local currently "bundle pristine" deletes the local gem's entire folder, and that's really not what I expected, I expected that the local folder would be skipped.

**Comment by aarontc** [comment]
I just ran into this minutes ago as well. Lost hours and hours of work on a gem I'm developing alongside a project which consumes the gem. Running Ruby 2.4.5 on Linux. Willing to provide more info if anything is needed to help resolve this.

**Comment by deivid-rodriguez** [comment]
I agree we should fix this.

**Comment by aarontc** [comment]
@colby-swandale I can understand that the intention of `pristine` is to clean up everything, but IMHO the entire point of having a local path override is to facilitate local development of a gem - in which case that local copy is, by definition, the pristine version already. I expected running `bundle pristine` would re-copy the local path into the local vendor/bundle cache so I'd get the changes I had made locally without having to bump the version number.

At the very least, I think a warning to the user with a countdown or confirmation would be much better than the current behavior of the system.

**Comment by colby-swandale** [comment]
I deleted my comment pretty much after i posted it. I can understand the perspective that you're coming from.

**Comment by baelter** [comment]
Just stepped on this mine and got my leg blown off :(
